### ISSUE TITLE
Multiple assets loading

### ISSUE THREAD

**Issue by vinicius73** [nextcloud/text#2451]
**Describe the bug**

Now is created one bundle for each "viewer" / "page".

- `main.js`
- `files.js`
- `public.js`
- `viewer.js`

Each of these files is bundled with Vue, Vuex, Nextcloud Components, Tiptap, CSS...
Although, the main issue could be the page load, it is causing other issues like "CSS collisions" and extra memory usage.

**Comment by max-nextcloud** [comment]
Isn't there a vendor.js that ships most of the dependencies such as Vue, Vuex, ...?

**Comment by juliushaertl** [comment]
Yep, most parts should be shared from the vendors or editor-*.js files:

```
-rw-r--r--  1 julius  staff    35K May 27 08:19 editor-collab.js
-rw-r--r--  1 julius  staff   6.3K May 27 08:19 editor-guest.js
-rw-r--r--  1 julius  staff    40K May 27 08:21 editor-rich.js
-rw-r--r--  1 julius  staff   201K May 27 08:21 editor.js
-rw-r--r--  1 julius  staff   1.0K May 27 08:21 files-modal.js
-rw-r--r--  1 julius  staff   213K May 27 08:21 text-files.js
-rw-r--r--  1 julius  staff   164K May 27 08:21 text-public.js
-rw-r--r--  1 julius  staff   121K May 27 08:21 text-text.js
-rw-r--r--  1 julius  staff    33K May 27 08:21 text-viewer.js
-rw-r--r--  1 julius  staff   2.5M May 27 08:21 vendors.js
```

But we can of course have a look if splitting like that still makes sense. The main reasoning behind the split and dynamic imports was that with the viewer we can only load the heavy parts if the user is actually opening the text app, e.g. on talk or the dashboard.


The text-public.js entrypoint we may drop once viewer is used on public pages of single shared files: https://github.com/nextcloud/viewer/issues/508

**Comment by vinicius73** [comment]
Yes, there is a vendor.
But for unknown reason not all libs are shared between bundles.

![image](https://user-images.githubusercontent.com/1561347/170744870-95637004-0a78-43e5-8f73-c7ae6257eb33.png)

**Comment by susnux** [comment]
A bit off topic: Maybe [ModuleFederation](https://webpack.js.org/plugins/module-federation-plugin/#specify-package-versions) of vendor packages (`vue`, `@nextcloud/...`) should be considered / discussed on a `nextcloud/core` (server) level to reduce traffic / improve loading times.

Even without any additional applications installed that would reduce the traffic of JS files about 10% (as currently `nextcloud/server` and `nextcloud/text` share many dependencies and `text` is enabled by default).
There might be 3rd party apps which want to use newer versions of some packages, but at least apps bound to nextcloud releases, like the `text` app, would benefit.

**Comment by vinicius73** [comment]
It is in my personal roadmap @susnux 
I wish I will be able to open some issues and PRs about it soon as possible.

**Comment by blizzz** [comment]
Are we still interesting in picking this up, at some point? It's been roughly a year since.
